packages/react-badge/src/components/CounterBadge/CounterBadge.types.ts

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { ComponentProps, ObjectShorthandProps } from '@fluentui/react-utilities';
+import { BadgeProps } from '../Badge/index';
+
+/**
+ * {@docCategory CounterBadge}
+ */
+export type CounterBadgeAppearance = 'filled' | 'ghost';
+
+/**
+ * {@docCategory CounterBadge}
+ */
+export type CounterBadgeShape = 'rounded' | 'circular';
+
+/**
+ * {@docCategory CounterBadge}
+ */
+export type CounterBadgeColors = 'accent' | 'warning' | 'important' | 'severe' | 'informative';
+
+/**
+ * {@docCategory CounterBadge}
+ */
+export interface CounterBadgeProps
+  extends ComponentProps,
+    React.HTMLAttributes<HTMLElement>,
+    Omit<BadgeProps, 'appearance' | 'shape'> {
+  /**
+   * A Badge can be square, circular or rounded
+   * @defaultvalue circular
+   */
+  shape?: CounterBadgeShape;
+
+  /**
+   * A Badge can be filled, outline, ghost, inverted
+   * @defaultvalue filled
+   */
+  appearance?: CounterBadgeAppearance;
+
+  /**
+   * A Badge can have color variations
+   * @defaultvalue accent
+   */
+  color?: CounterBadgeColors;
+
+  /**
+   * Max number to be displayed
+   * @defaultvalue 99
+   */
+  overflowCount?: number;
+
+  /**
+   * Value diplayed by the Badge
+   * @defaultvalue 0
+   */
+  count?: number;
+
+  /**
+   * If 0 number should be displayed
+   * @defaultvalue true
+   */
+  showZero?: boolean;
+}
+
+/**
+ * {@docCategory CounterBadge}
+ */
+export interface CounterBadgeState extends CounterBadgeProps {
```

Comment:
This could extend BadgeState, and only specify the additional defaulted props here.

Reply:
I don't think so since `appearance` and `shape` from `BadgeState` are going to be different. We are going to end up with things like:

```typescript
export interface CounterBadgeState extends Omit<CounterBadgeProps, 'icon'>, Pick<BadgeState, 'ref' | 'icon'> {

// or

export interface CounterBadgeState extends Omit<BadgeState, 'appearence' | 'shape'>, Pick<CounterBadgeProps, 'appearance' | 'shape'> {

```

Reply:
Looking at the code, it seems that `CounterBadgeProps` is already extending `BadgeProps` and omitting what isn't necessary. I think some additional `Extract` could be used to reuse `BadgeProps` types.

However, extending `BadgeState` wouldn't necessarily make sense since the types `BadgeState` would conflict with the narrowed types in `CounterBadgeProps`  ?

Reply:
Fixed